spec/transactions/todos/create_spec.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+RSpec.describe Transactions::Todos::Create do
+  let(:todo_repository) { instance_double('TodoRepository') }
+  let(:todo) { create(:todo, name: "Task1") }
+
+  before do
+    allow(todo_repository).to receive(:build).with(input).and_return(todo)
+    allow(todo_repository).to receive(:save).with(todo).and_return(true)
+  end
+
+  context 'when input is valid' do
+    subject(:transaction) { described_class.new(todo_repository: todo_repository) }
+    let(:input) { { name: 'Task1', completed: false } }
+
+    it 'returns success with the todo' do
+      result = transaction.call(input)
+      expect(result).to be_success
+      expect(result.value!).to eq(todo)
+    end
+  end
+
+  context 'when input is invalid' do
+    let(:input) { { completed: false } }
+    subject(:transaction) { described_class.new }
+
+    it 'returns failure' do
+      result = transaction.call(input)
+      expect(result).to be_failure
+      expect(result.failure).to include(:name)
+    end
```

Comment:
This test would be more fit for the create Contract, we should test only the behaviour of the transaction

Let's sync tomorrow on how we can write these tests